sympy/simplify/simplify.py

```diff
@@ -868,6 +873,29 @@ def _nthroot_solve(p, n, prec):
                 return sol
 
 
+def quantity_simplify(expr):
+    from sympy.physics.units import Quantity, length
+    n = len(expr.args)
+    a = 1
```

Comment:
I would recommend a defaultdict and collecting by dimension the arguments.

Plus, you should treat only Mul like this.